### PR TITLE
p2p: Clarify control flow in ProcessMessage

### PR DESCRIPTION
**Pull request by maflcko**
`ProcessMessage` is effectively a massive switch case construct. In the past there were attempts to clarify the control flow in `ProcessMessage()` by moving each case into a separate static function (see #9608). It was closed because it wasn't clear if moving each case into a function was the right approach.
Though, we can quasi treat each case as a function by adding a return statement to each case. (Can be seen as a continuation of bugfix #13162)

This patch does exactly that.

Also note that this patch is a subset of previous approaches such as #9608 and #10145.

Review suggestion: `git diff HEAD~ --function-context`

### CODE REVIEW

**Comment by MarcoFalke** [comment]
As a nice side effect, bugs such as  #13162 are now impossible.

**Comment by practicalswift** [comment]
utACK fa37c5b662f1aeeaa53e60377979156e84d550b4

Clarification is good

**Comment by domob1812** [comment]
utACK fa37c5b662f1aeeaa53e60377979156e84d550b4

Seems like a reasonable clarification with minimal changes.

**Comment by sipa** [comment]
Any reason not to drop all the `else`s in between the different options?

**Comment by MarcoFalke** [comment]
Made everything early-return as requested by @sipa

**Comment by Empact** [comment]
Not necessarily to be fixed in this PR, but I noticed that due to `!fImporting && !fReindex`, `CMPCTBLOCK`, `BLOCKTXN`, `HEADERS`, and `BLOCK` messages will be treated with
`LogPrint(BCLog::NET, "Unknown command \"%s\" from peer=%d\n"` during import or reindex. The failure could be more informative.

**Comment by practicalswift** [comment]
utACK fa6c3dea420b6c50c164ccc34f4e9e8a7d9a8022

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2384,10 +2374,10 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 Misbehaving(pfrom->GetId(), nDoS);
             }
         }
+        return true;
     }
 
-
-    else if (strCommand == NetMsgType::CMPCTBLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
+    if (strCommand == NetMsgType::CMPCTBLOCK && !fImporting && !fReindex) // Ignore blocks received while importing
```
Comment (jnewbery) [inline src/net_processing.cpp]: May be out of scope for this PR, but I think it'd be better for these conditionals to be nested:

```c++
if (strCommand == NetMsgType::CMPCTBLOCK) {
    if (!fImporting && !fReindex) { // Ignore blocks received while importing
        ...
    }
    // could optionally log in an else branch here
    return true;
}
```

for a couple of reasons:

- it prevents compact blocks from being logged as "Unknown command" if received during import/reindex
- it maintains your new control flow of each message type returning after its own code block.

Same for BLOCKTXN, HEADERS and BLOCK message code blocks below.

**Review by jnewbery — COMMENTED** [review]
utACK fa6c3dea420b6c50c164ccc34f4e9e8a7d9a8022

one comment inline.

File: src/net_processing.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/net_processing.cpp]: Indeed out of scope, as this is not refactoring, but bugfixing-like territory.

Comment (jnewbery) [reply src/net_processing.cpp]: ok

**Comment by laanwj** [comment]
I'm not sure this is **much** better with the early returns. though, this does untangle the case and makes it easier to split up the function and move to dispatching based on say, a hash table, in the future, so it's a step in the right direction.

utACK fa6c3dea420b6c50c164ccc34f4e9e8a7d9a8022

**Comment by JeremyRubin** [comment]
post-merge utack, good incremental step.

**Pull request merged**